compiler/msgs.nim

```diff
@@ -64,6 +64,8 @@ type
     errVarForOutParamNeeded,
     errPureTypeMismatch, errTypeMismatch, errButExpected, errButExpectedX,
     errAmbiguousCallXYZ, errWrongNumberOfArguments,
+    errWrongNumberOfArgumentsInCall,
```

Comment:
I have a refactoring proposal for dealing with error messages, but not enough time to implement it.

My idea is to define a helper in msgs.nim, which will allow you to say:

``` Nim
error errXRequiresOneArgument "$1 requires one parameter"
```
This will automatically define a template that you can use in the rest of the code like this:

``` Nim
errXRequiresOneArgument "foo"
```

Under the hood, this may even revive your initial vision of a localized compiler :)